base/deprecated.jl

```diff
@@ -1740,6 +1740,8 @@ end
 @deprecate mapfoldl(f, op, v0, itr) mapfoldl(f, op, itr; init=v0)
 @deprecate mapfoldr(f, op, v0, itr) mapfoldr(f, op, itr; init=v0)
 
+@deprecate startswith(a::Vector{UInt8}, b::Vector{UInt8}) view(a, 1:length(b)) == b
```

Comment:
This throws a `BoundsError` when `length(b) > a` while the method it deprecates return `false`. Or?

Reply:
Indeed. I've added a check. It's not pretty but that method is probably not used often anyway.